controller/src/plan.rs

```diff
@@ -45,9 +46,25 @@ impl ControllerPlan {
                 None
             };
 
+            let domain_name = options
+                .domain_name
+                .and_then(|ref name| {
+                    Name::from_utf8(name)
+                        .or_else(|e| {
```

Comment:
I find that doing chaining inside of another chain gets pretty hard to read, maybe this could be expressed better in another way? At least for now, I'd take clippy's suggestion and use `map_err`.

Reply:
yea, I tried using anyhow and a context, but that was even trickier.

forgot about map_err though